mkdocs/docs/expression-dsl.md

```diff
@@ -0,0 +1,259 @@
+age_field = Reference("age")
+```
+
+### Predicates
+
+Predicates are expressions that evaluate to a boolean value. They can be combined using logical operators.
+
+#### Literal Predicates
+
+```python
+from pyiceberg.expressions import EqualTo, NotEqualTo, LessThan, LessThanOrEqual, GreaterThan, GreaterThanOrEqual
+
+# age equals 18
+age_equals_18 = EqualTo("age", 18)
+
+# age is not equal to 18
+age_not_equals_18 = NotEqualTo("age", 18)
+
+# age is less than 18
+age_less_than_18 = LessThan("age", 18)
+
+# Less than or equal to
+age_less_than_or_equal_18 = LessThanOrEqual("age", 18)
+
+# Greater than
+age_greater_than_18 = GreaterThan("age", 18)
+
+# Greater than or equal to
+age_greater_than_or_equal_18 = GreaterThanOrEqual("age", 18)
+```
+
+#### Set Predicates
+
+```python
+from pyiceberg.expressions import In, NotIn
+
+# age is one of 18, 19, 20
+age_in_set = In("age", [18, 19, 20])
+
+# age is not 18, 19, oer 20
+age_not_in_set = NotIn("age", [18, 19, 20])
+```
+
+#### Unary Predicates
+
+```python
+from pyiceberg.expressions import IsNull, NotNull
+
+# Is null
+name_is_null = IsNull("name")
+
+# Is not null
+name_is_not_null = NotNull("name")
+```
+
+#### String Predicates
+
+```python
+from pyiceberg.expressions import StartsWith, NotStartsWith
+
+# TRUE for 'Johnathan', FALSE for 'Johan'
+name_starts_with = StartsWith("name", "John")
+
+# FALSE for 'Johnathan', TRUE for 'Johan'
+name_not_starts_with = NotStartsWith("name", "John")
+```
+
+### Logical Operators
+
+You can combine predicates using logical operators:
+
+```python
+from pyiceberg.expressions import And, Or, Not
+
+# TRUE for 25, FALSE for 67 and 15
+age_between = And(
+    GreaterThanOrEqual("age", 18),
+    LessThanOrEqual("age", 65)
+)
+
+# FALSE for 25, TRUE for 67 and 15
+age_outside = Or(
+    LessThan("age", 18),
+    GreaterThan("age", 65)
+)
+
+# NOT operator
+not_adult = Not(GreaterThanOrEqual("age", 18))
+```
+
+## Advanced Usage
+
+### Complex Expressions
+
+You can build complex expressions by combining multiple predicates and operators:
+
+```python
+from pyiceberg.expressions import And, Or, Not, EqualTo, GreaterThan, LessThan, In
+
+# (age >= 18 AND age <= 65) AND (status = 'active' OR status = 'pending')
+complex_filter = And(
+    And(
+        GreaterThanOrEqual("age", 18),
+        LessThanOrEqual("age", 65)
+    ),
+    Or(
+        EqualTo("status", "active"),
+        EqualTo("status", "pending")
+    )
+)
+
+# NOT (age < 18 OR age > 65)
+age_in_range = Not(
+    Or(
+        LessThan("age", 18),
+        GreaterThan("age", 65)
+    )
+)
+```
+
+### Type Safety
+
+The expression DSL provides type safety through Python's type system. When you create expressions, the types are checked at runtime:
+
+```python
+from pyiceberg.expressions import EqualTo
+
+# This will work
+age_equals_18 = EqualTo("age", 18)
+
+# This will raise a TypeError if the field type doesn't match
+age_equals_18 = EqualTo("age", "18")  # Will fail if age is an integer field
```

Comment:
This should work:

![image](https://github.com/user-attachments/assets/9323ef86-e1d1-4319-bbcf-39a783c6d005)